### PR TITLE
Finalize fixes on WIndows test suite for v0.1.1

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Towards a fully working Windows test suite! Several of our unit tests (~8) have been failing since #434, and we've addressed them in several PRs already. This one will hopefully squash the last of them.

Merge after these are completed (in order):
- [x] Fix `logo` image discrepancy with conda ghostscript (#434)
- [x] Fix Windows not finding 'share' directory (#437, and related upstream fix)
- [x] Fix Windows on Azure Pipelines not connecting to http://oceania.generic-mapping-tools.org/ (#438)
- [x] Fix Windows not finding `gmt.dll` without `GMT_LIBRARY_PATH` environment variable (#440)
- [x] (This PR)
  - [x] Fix sphinx pygmtscraper setting temporary directory on different mount paths (590a523dcb49fbff426fb08c2de9d8251ee47d34)
  - [x] Fix pandas.Series returning dtype=int64 in Windows repr (efe35ec760332438081d5a12ac113c3b33d04227)
  - [x] Fix issues with c_int/c_long type (92029417db518ed3ed51a4ba823a242328f2ccc8)

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

**Comment by weiji14** [comment]
Last 2 tests to fix! Investigating now, might be related to how `numpy` on Windows handles integers compared to Linux/MacOS. (see e.g. https://github.com/numpy/numpy/issues/10317?)

```python-traceback
================================== FAILURES ===================================
__________________ [doctest] pygmt.clib.conversion._as_array __________________
221 
222     Examples
223     --------
224 
225     >>> import pandas as pd
226     >>> x_series = pd.Series(data=[1, 2, 3, 4])
227     >>> x_array = _as_array(x_series)
228     >>> type(x_array)
229     <class 'numpy.ndarray'>
230     >>> x_array
Expected:
    array([1, 2, 3, 4])
Got:
    array([1, 2, 3, 4], dtype=int64)

C:\Miniconda\envs\testing\Lib\site-packages\pygmt\clib\conversion.py:230: DocTestFailure
___________ [doctest] pygmt.clib.conversion.kwargs_to_ctypes_array ____________
263     Returns
264     -------
265     ctypes_value : ctypes array or None
266 
267     Examples
268     --------
269 
270     >>> import ctypes as ct
271     >>> value = kwargs_to_ctypes_array('bla', {'bla': [10, 10]}, ct.c_int*2)
272     >>> type(value)
Expected:
    <class 'pygmt.clib.conversion.c_int_Array_2'>
Got:
    <class 'pygmt.clib.conversion.c_long_Array_2'>

C:\Miniconda\envs\testing\Lib\site-packages\pygmt\clib\conversion.py:272: DocTestFailure
```

**Comment by seisman** [comment]
Perhaps change the tests a little bit. For example, changing
```
>>> x_array
```
to 
```
>>> x_array.dtype

>>> list(x_array)
```

File: pygmt/clib/conversion.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def _as_array(vector):
 
     """
     if isinstance(vector, pandas.Series):
-        return vector.values
+        return vector.to_numpy()
```
Comment (weiji14) [inline pygmt/clib/conversion.py:244]: Actually, can't we just get rid of this `_as_array()` function?  Calling `np.asarray` on a `pandas.Series` returns a `numpy.array` anyway right? I don't see why this is even here.

Comment (seisman) [reply pygmt/clib/conversion.py:244]: Yes, looks good to me. Not sure why `_as_array()` was designed like this.

Comment (weiji14) [reply pygmt/clib/conversion.py:244]: On another note, see https://www.pypandas.cn/en/blog/pandas-extension-arrays.html#the-current-state for why the change from `.values` to `.array` or `.to_numpy` has been made in Pandas.

Comment (weiji14) [reply pygmt/clib/conversion.py:244]: Looking at `git blame`, this code goes all the way back to December 2017 in #100. Back when pandas was at v0.21.

Comment (weiji14) [reply pygmt/clib/conversion.py]: Cool, I've remove the `_as_array()` function in favour of numpy's built-in `.asarray`.

File: pygmt/clib/conversion.py [hunk 2]
```diff
@@ -268,9 +226,9 @@ def kwargs_to_ctypes_array(argument, kwargs, dtype):
     --------
 
     >>> import ctypes as ct
-    >>> value = kwargs_to_ctypes_array('bla', {'bla': [10, 10]}, ct.c_int*2)
+    >>> value = kwargs_to_ctypes_array('bla', {'bla': [10, 10]}, ct.c_long*2)
     >>> type(value)
-    <class 'pygmt.clib.conversion.c_int_Array_2'>
+    <class 'pygmt.clib.conversion.c_long_Array_2'>
```
Comment (weiji14) [inline pygmt/clib/conversion.py:231]: `c_int` is just an alias for `c_long` according to https://docs.python.org/3/library/ctypes.html#ctypes-tutorial. Changing it to `c_long` should make this test pass in a cross-platform way.

**Comment by weiji14** [comment]
Awesome, all green and good to go!

**Pull request merged**